### PR TITLE
Markdown Renderer

### PR DESCRIPTION
**Pull request by chamod-gamage**
## Notion ticket link

[Markdown Renderer](https://www.notion.so/uwblueprintexecs/Markdown-Render-Content-Box-cc263d9b98474382ab27e97ddb08ba6f)

## Implementation description
* Implemented all the functionality described [here](https://www.notion.so/uwblueprintexecs/Markdown-Editor-990ac8a8782b4ea28b272497319a794c) except alignment (see Covered by Renderer column)

## Steps to test
1. Visit `localhost:3000/editor`

Once approved, I'll delete the /editor page since it's only there for testing purposes

## What should reviewers focus on?
* Whether the editor works as expected and is formatted nicely

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: src/components/markdownRenderer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                            <a href={rest.href}>{children}</a>
+                        )}
+                    </>
+                ),
+            }}
+            className={style.markdown + " right"}
+        >
+            {content.replace("$[", "[")}
+        </ReactMarkdown>
+        {content}
```
Comment (Aphexis) [inline src/components/markdownRenderer/index.tsx:48]: Is this necessary or was it just in for debugging?

also, do we need the `md-row` div wrapping each of the lines? We could maybe just have a div wrapping the entire markdown output, but I think each ReactMarkdown line already has its own div

File: src/components/markdownRenderer/markdown-renderer.module.css [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+.markdown pre {
+    font: inherit;
+}
+
+.markdown code {
+    white-space: pre;
+    text-decoration: none !important;
+    text-decoration-color: white;
+}
```
Comment (Aphexis) [inline src/components/markdownRenderer/markdown-renderer.module.css:55]: do we need/use the `code` formatting?

**Review by Aphexis — COMMENTED** [review]
This is really awesome!
Checked over the commands, noticed some edge-case things:

- underline is doing the same thing as bolding text, I think we need custom formatting there
- hyperlinks are clickable but don't have any formatting that distinguishes them from text (not sure how it should appear). Also, links without `https://` prefixing the url will link relative to localhost (maybe fine, just worth noting)
- Is it meant to be possible to make indented bulleted lists? I noticed that if I have `- - text`, I'll get an indented bullet but the outer bullet is also still there (so the line has two bullets)
- I noticed when you hit tab, the format for the line becomes monospace/code, not sure if that's intentional (I assume this is from the renderer library?)
- I noticed that H6 is bigger than H5, not sure if that's how it's meant to be rendered?

(also saw that there were some self-check and other commits/changes mixed into this PR, maybe we can take those out if that's easy to do)

File: src/components/markdownRenderer/index.tsx (hunk 1 above)
Comment (chamod-gamage) [reply src/components/markdownRenderer/index.tsx:48]: 1) U right mb
2) Basically I had to put each MD row on its own div with {flex-direction: row} to make sure the styling of components on each line was correct even with $ to indent, while also laying out the components on a given line left to right. I think we may have to keep it even tho it's not the most elegant solution

File: src/components/markdownRenderer/markdown-renderer.module.css (hunk 2 above)
Comment (chamod-gamage) [reply src/components/markdownRenderer/markdown-renderer.module.css:55]: yea so basically 'code' isn't actually the monospace stuff you would expect, it's what I use to be able to process indentation via $

Essentially, white space is usually reduced to a single space by the DOM (there was a good reason but don't remember why), so I used the 'code' styling to force whitespace

**Comment by chamod-gamage** [comment]
TODO:

- [ ] Styling for hyperlinks
- [ ] Indentation fix
- [ ] Underline differentiation

**Comment by tawnyzhao** [comment]
Heading issue:
image

**Review by tawnyzhao — APPROVED** [review]
approve - leave todo's in Notion

**Review by yashgajaria — APPROVED** [review]
approved, tested it out and all necessary todo's are already documented.

**Pull request merged**